pkg/acsengine/munge.go

```diff
@@ -0,0 +1,194 @@
+package acsengine
```

Comment:
To me, those sound like generic C#-y names. I've not really seen Go files names like that, especially given that they're more groupings of functionality than C# where it's usually 1:1 class<->file.

I'm open to something beyond "munge" but it seems more descriptive than "TemplateProcessor" to me. "TemplateProcessor" seems more like the stuff we have in `engine.go`.